File: #17/JesusWay69.py
import os, random
os.system('cls')

"""* EJERCICIO:
 * Utilizando tu lenguaje, emplea 3 mecanismos diferentes para imprimir
 * números del 1 al 10 mediante iteración.
 *
 * DIFICULTAD EXTRA (opcional):
 * Escribe el mayor número de mecanismos que posea tu lenguaje
 * para iterar valores. ¿Eres capaz de utilizar 5? ¿Y 10?"""

'''FOR CON RANGO CONDICIONAL DE INICIO Y PARADA'''
for i in range (1,11):
    print(i, end=" ")
i=0
print()

'''FOR CON RANGO SOLO DE PARADA Y CONDICIONAL PARA SALTARSE EL 0 CON EL QUE EMPIEZA POR DEFECTO'''
for i in range (11):
    if i == 0:
        continue
    print(i, end=" ")
i=0
print()
'''FOR ANIDADOS CON CONDICIÓN DE INICIO,PARADA Y PASO, EL PRIMERO IMPRIME LOS IMPARES Y EL SEGUNDO LOS PARES'''
for i in range (1,11,2):
    print(i, end=" ")
    for j in range(1):
        if i<10:
            print (i+1, end=" ")
i=0
print() 
'''BUCLE FOR-ELSE CON CONDICIÓN DE INICIO Y PARADA'''
for i in range (0,1):
    if i == 1:
        break 
else:
    for i in range(i+1,11):
        print(i,end=" ")
i=0
print()

   
'''WHILE CON CONDICIÓN DE PARADA Y OPERADOR DE INCREMENTO'''
while i<10:
    i+=1
    print(i, end=" ")
i=0
print()

'''WHILE CON OPERADOR DE INCREMENTO Y CONDICIÓN DE SALIDA (DO-WHILE)'''
while True:
    i+=1
    print (i,end=" ")
    if i == 10:
        break
i=0
print()

'''WHILE ANIDADOS CON CONDICIÓN DE PARADA Y OPERADOR DE INCREMENTO'''
while i<10:
   i+=1
   print(i, end=" ")
   while i<10:
       i+=1
       print(i, end=" ")
i=0
print()

'''FUNCIÓN CON INCREMENTO, CONDICIÓN DE PARADA Y RECURSIVIDAD'''
def iterator_1(i:int):
    i+=1
    print(i, end=" ")
    if i<=9:
        iterator_1(i)
iterator_1(i)
print()

'''FUNCIÓN CON GENERACIÓN DE NÚMEROS ALEATORIOS QUE SE AÑADEN A UN SET HASTA QUE ESTE CUMPLA LA CONDICIÓN DE LONGITUD 10'''
def iterator_2(i:int):
    my_set = set()
    while len(my_set)<i:
       num = random.randint(1,10)
       my_set.add(num)
    [print(num, end=" ") for num in my_set]
iterator_2(10)
print()

'''FUNCIÓN CON MÉTODO DE LENGUAJE ITER'''
def iterator_3():
    my_list = []
    [my_list.append(i) for i in range(1,11)]
    my_iterator = (iter(my_list))
    [print(next(my_iterator), end=" ") for i in range (len(my_list))]
iterator_3()
print()

def iterator_4(symbols:str):
    for char in symbols:
        print(ord(char)-40,end=" ")
iterator_4(")*+,-./012")

File: #17/JesusWay69.java
package roadmap.ejercicio_17;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;
import java.util.TreeSet;

/*
 * EJERCICIO:
 * Utilizando tu lenguaje, emplea 3 mecanismos diferentes para imprimir
 * números del 1 al 10 mediante iteración.
 *
 * DIFICULTAD EXTRA (opcional):
 * Escribe el mayor número de mecanismos que posea tu lenguaje
 * para iterar valores. ¿Eres capaz de utilizar 5? ¿Y 10?
 */
public class JesusWay69 {

    public static void main(String[] args) {
        iterator1();
        iterator2();
        iterator3();
        iterator4();
        iterator5();
        iterator6();
        iterator7();
        iterator8();
        iterator9(0);
        iterator10();
        iterator11();
        iterator12();
        iterator13();
        iterator14();
    }

    private static void iterator1() {
        for (int i = 1; i <= 10; i++) {
            System.out.print(i + " ");
        }
        System.out.print(" --> Método 1");
        System.out.println("");
    }

    private static void iterator2() {
        int i = 1;
        while (i <= 10) {
            System.out.print(i + " ");
            i++;
        }
        System.out.print(" --> Método 2");
        System.out.println("");
    }

    private static void iterator3() {
        int i = 1;
        do {
            System.out.print(i + " ");
            i++;
        } while (i <= 10);
        System.out.print(" --> Método 3");
        System.out.println("");
    }

    private static void iterator4() {
        for (int i = 1, j = 2; i < 10 && j <= 10; i += 2, j += 2) {//for con 2 índices de control(par,impar)
            System.out.print(i + " " + j + " ");
        }
        System.out.print(" --> Método 4");
        System.out.println("");
    }

    private static void iterator5() {
        for (int i = 1, j = 2; i < 10 && j <= 10; i++, j++) {
            if (i % 2 != 0 || j % 2 == 0) {
                System.out.print(i + " " + j + " ");
            }
        }
        System.out.print(" --> Método 5");
        System.out.println("");
    }

    private static void iterator6() {
        int i;
        for (i = 1; i <= 5; i++) {
            if (i <= 5) {
                System.out.print(i + " ");
                continue;
            }
        }
        for (int j = i; j <= 10; j++) {
            System.out.print(j + " ");
        }
        System.out.print(" --> Método 6");
        System.out.println("");
    }

    private static void iterator7() {
        int i = 1;
        for (; true;) {
            if (i <= 10) {
                System.out.print(i + " ");
                i++;
            } else {
                System.out.print(" --> Método 7");
                System.out.println("");
                break;
            }

        }
    }

    private static void iterator8() {
        int i;
        for (i = 1; i <= 10; i += 2) {
            System.out.print(i + " ");
            for (int j = 0; j < 1; j++) {
                if (i < 10) {
                    System.out.print(i + 1 + " ");
                }
            }
        }
        System.out.print(" --> Método 8");
        System.out.println("");
    }

    private static void iterator9(int i) {
        if (i < 10) {
            System.out.print(++i + " ");
            iterator9(i);
        } else {
            System.out.print(" --> Método 9");
            System.out.println("");
        }

    }

    private static void iterator10() {
        StringBuilder nums = new StringBuilder();
        for (int i = 1; i <= 10; i++) {
            nums.append(i);
        }
        for (int j = 0; j <= nums.length() - 1; j++) {
            if (nums.charAt(j) == '0') {
                System.out.print("\b0 ");
            } else {
                System.out.print(nums.charAt(j) + " ");
            }
        }
        System.out.print(" --> Método 10");
        System.out.println("");

    }

    private static void iterator11() {
        Set<Integer> mySet = new TreeSet<Integer>();
        do {
            int num = (int) (Math.random() * 10 + 1);
            mySet.add(num);
        } while (mySet.size() <= 9);

        for (Integer value : mySet) {
            System.out.print(value + " ");
        }
        System.out.print(" --> Método 11");
        System.out.println("");
    }

    private static void iterator12() {
        char[] myList = {'3', '4', '5', '6', '7', '8', '9', ':', ';', '<'};

        for (int myCode : myList) {
            System.out.print((myCode - 50) + " ");
        }
        System.out.print(" --> Método 12");
        System.out.println("");

    }

    private static void iterator13() {
        int i = 0;
        while (i < 2) {
            i++;
            System.out.print(i + " ");
            while (i < 3) {
                i++;
                System.out.print(i + " ");
                while (i < 4) {
                    i++;
                    System.out.print(i + " ");
                    while (i < 5) {
                        i++;
                        System.out.print(i + " ");
                        while (i < 10) {
                            i++;
                            System.out.print(i + " ");
                        }
                    }
                }
            }
        }
        System.out.print(" --> Método 13");
        System.out.println("");
    }

    private static void iterator14() {
        List<String> myList = new ArrayList<>();
        int a = 1, b = 11, c = b + a, d = b * 2, e = c + d, f = e + d, g = f + d, h = g / 2 - 30, i = (g + d) / 10;
        myList.add(c + "");
        myList.add(e + "");
        myList.add(f + "");
        myList.add(g + "");
        myList.add(h + "");
        myList.add(i + "");

        for (String element : myList) {
            for (int j = 0; j < element.length(); j++) {
                if (element.charAt(j) == '0') {
                    System.out.print("\b");
                }
                System.out.print(element.charAt(j) + " ");
            }
        }
        System.out.print(" --> Método 14");
        System.out.println("");
    }

}
